beta/experience-platform-extension/tutorials/tutorial-4-rules-xdm-events.md

```diff
@@ -0,0 +1,154 @@
+
+#### Swift
+
+Download the iOS Swift Sample application from [https://github.com/adobe/aepsdk-sample-app-ios/tree/beta-assignment-4](https://github.com/adobe/aepsdk-sample-app-ios/archive/beta-assignment-4.zip).
+
+To get started, follow the steps described in [AEP SDK Sample App Swift - Installation](https://github.com/adobe/aepsdk-sample-app-ios/tree/beta-assignment-4#installation).
+{% endtab %}
+{% endtabs %}
+
+### Set up the required fields
+
+**Note:** The steps in this section are not required if you already implemented the steps described in [Assignment 3 - XDM implementation](https://aep-sdks.gitbook.io/docs/beta/experience-platform-extension/tutorials/tutorial-4-rules-xdm-events).
+
+In [Adobe Experience Platform Launch](https://experience.adobe.com/launch), go to the **Environments** tab in the mobile property created in Assignment 1 and click on the Development![img](https://firebasestorage.googleapis.com/v0/b/gitbook-28427.appspot.com/o/assets%2F-Lf1Mc1caFdNCK_mBwhe%2F-Lf1N06T8hdv0-r5jPPN%2F-Lf1N3-ofPO9fLFT1edw%2Fscreen-shot-2018-10-18-at-11.22.17-am.png?generation=1558039279051937&alt=media)icon. Find the Environment File ID at the top and copy it.
+
+Set the `LAUNCH_ENVIRONMENT_FILE_ID` to the copied Environment File ID in the `MainApp` \(Android\) / `AppDelegate` \(iOS\) file.
+
+Set the `PRODUCT_REVIEW_DATASET_ID` to the dataset identifier for Product reviews created in Assignment 3 in the `EdgeTab` (Android) / `EdgeViewController` (iOS) file.
+
+In the same file, set the `TENANT_ID` to the org identifier as described in the Product reviews schema.
+
+## Attach data to XDM events
+
+### Update the Product reviews schema
+
+1. In the browser, navigate to [Adobe Experience Platform](https://experience.adobe.com/platform) and login with your credentials.
+
+2. Update the `Product reviews` schema created in Assignment 3 by adding two new fields.
+
+   - From the left panel, select `Schemas` and then click on the `Product reviews` schema you have created for the previous assignment.
+
+   - Click on the `Product review` mixin.
+
+   - Click on the plus icon `+` next to the schema name and add the following fields. Click `Apply` after each field added.
+
+     | Field name   | Display name  | Type   | Required |
+     | :----------- | :------------ | :----- | -------- |
+     | ratingSource | Rating Source | String | No       |
+     | ratingDate   | Rating Date   | String | No       |
+
+   - Click `Save` to update the schema.
+
+### Create a mobile rule in Adobe Experience Launch
+
+1. Navigate to [Adobe Experience Launch](https://experience.adobe.com/launch) and select the mobile property you created in the previous assignments.
+
+2. Click on `Extensions` from the left panel and update the `Adobe Experience Platform Edge` extension to latest version.
+
+3. Click on `Data elements` from the left panel and the following data elements:
+
+   1. Create `App ID` data element:
+      - Click on the `Create New Data Element` button.
```

Comment:
"`Add Data Element`"

Reply:
First time you create a data element it shows as `Create New Data Element`.

Reply:
Ah, ignore comment. "Create New Data Element" is correct when there are no existing data elements. :)